### ISSUE TITLE
Native exception is thrown when calling requestPurchase after upgrading to 10.0.1

### ISSUE THREAD

**Issue by yanush** [dooboolab/react-native-iap#1898]
Native exception is thrown when calling requestPurchase after upgrading to 10.0.1

I'm trying to upgrade my library to v10.0.1 and I can't solve this issue. It happens when using the iOS emulator

- react-native-iap: 10.0.1
- react-native: 0.69.4
- Platforms (iOS, Android, emulator, simulator, device): iOS emulator

this is my method:

```
    async requestPurchase(productId) {
        try {
            const result = await requestPurchase({ productId });
            console.log('purchased successfully');
        } catch (error) {
            console.log('could not purchase');
        }
    }

```
Attached a screenshot of the exception.
Please advise

![Simulator Screen Shot - iPhone 12 - 2022-08-30 at 10 18 37](https://user-images.githubusercontent.com/2669268/187374991-a4927d2f-eb57-46f2-8534-aa3f27bcfc7f.png)

**Comment by yanush** [comment]
I have tried it in every possible way
This is what I get when I use `await requestPurchase({ sku: productId });`:
![Simulator Screen Shot - iPhone 12 - 2022-08-31 at 10 16 55](https://user-images.githubusercontent.com/2669268/187616892-9af9533d-eadf-4807-a229-38c11e85672e.png)

**Comment by andresesfm** [comment]
@yanush are you sure you ran `yarn`? because the parameter name `appAccountToken` is no longer in the code base. I'm thinking that it could be a regression but it could also be that you are not using the version you think. Please let us know

**Comment by hkeithk** [comment]
@andresesfm im seeing a similar issue here only on android devices. I am on rn-iap version 10.0.1, react native 0.68.2. platforms: android only

Unfortunately when the error occurs, I dont see any error message provided in the logs. Strangely enough, the google play store dialog opens, while the app crashes in the background. 

![image](https://user-images.githubusercontent.com/22736937/187800725-21b81902-bd91-4e33-9e4c-12fc9f3bdacd.png)

**Comment by andresesfm** [comment]
@hkeithk This is for an iOS issue please create a separate issue using the provided template

**Comment by andresesfm** [comment]
@yanush please try `requestPurchase({ sku:productId });`

**Comment by yanush** [comment]
Seems like it was fixed in the latest version (10.0.5)
Thanks
